src/ring_theory/ideal/basic.lean

```diff
@@ -84,16 +78,6 @@ theorem eq_top_iff_one : I = ⊤ ↔ (1:α) ∈ I :=
 theorem ne_top_iff_one : I ≠ ⊤ ↔ (1:α) ∉ I :=
 not_congr I.eq_top_iff_one
 
-lemma exists_mem_ne_zero_iff_ne_bot : (∃ p ∈ I, p ≠ (0 : α)) ↔ I ≠ ⊥ :=
```

Comment:
At a glance, I'm surprised this isn't true for semirings

Reply:
Eric, at the moment, `ideal` is only defined for a `comm_semiring`.  In fact, below is a proof (the first one that I found) of the "same" result for a submodule of `R`, when `R` is a `semiring`.  I think that this should be a separate PR, in case it is not already there.

```
lemma submodule.ne_top_iff_one {R : Type*} [semiring R]
  (J : submodule R R): J ≠ ⊤ ↔ (1:R) ∉ J :=
not_congr ⟨λ a, submodule.eq_top_iff'.mp a 1, λ hx, submodule.ext (λ x, ⟨λ _, submodule.mem_top, λ xj, by { rw [← mul_one x], exact submodule.smul_mem J x hx }⟩)⟩
```